src/Downloader.ts

```diff
@@ -184,45 +182,61 @@ class Downloader {
-    // check if RESTBase-powered API available
-    try {
-      const restApiMainPageQuery = await this.getJSON<any>(`${this.baseUrl}${encodeURIComponent(this.mw.metaData.mainPage)}`);
-      this.mwCapabilities.restApiAvailable = !!restApiMainPageQuery.lead;
-    } catch (err) {
-      this.mwCapabilities.restApiAvailable = false;
-      logger.warn(`Failed to get remote Rest API`);
+  public async setBaseUrls() {
+    const isVeApiAvailable = this.mwCapabilities.veApiAvailable;
+    const isMobileApiAvailable = this.mwCapabilities.mobileRestApiAvailable;
+    const isDesktopApiAvailable = this.mwCapabilities.desktopRestApiAvailable;
+
+    // Setting the default values
+    this.baseUrlForMainPage = this.mw.veApiUrl.href;
+    this.baseUrl = this.mw.veApiUrl.href;
+
+    // Check for desktop API
+    if(isDesktopApiAvailable) {
+      this.baseUrl = this.mw.desktopRestApiUrl.href;
     }
 
-    if (!this.forceLocalParser) {
-      // check if VisualEditor available
-      try {
-        const parsoidMainPageQuery = await this.getJSON<any>(`${this.mw.veApiUrl.href}${encodeURIComponent(this.mw.metaData.mainPage)}`);
-        this.mwCapabilities.veApiAvailable = !!parsoidMainPageQuery.visualeditor.content;
-      } catch (err) {
-        this.mwCapabilities.veApiAvailable = false;
-        logger.warn(`Failed to get remote Parsoid`);
-      }
+    // Check for Mobile API
+    if(isMobileApiAvailable) {
+      this.baseUrl = this.mw.mobileRestApiUrl.href;
     }
 
+    // Setting Values for custom flag: noLocalParserFallback
     if (!this.noLocalParserFallback) {
-      if (!this.mwCapabilities.restApiAvailable || !this.mwCapabilities.veApiAvailable) {
-        logger.log(`Using local MCS and ${this.mwCapabilities.veApiAvailable ? 'remote' : 'local'} Parsoid`);
+      if (!isVeApiAvailable) {
         await this.initLocalServices();
 
-        if (!this.mwCapabilities.restApiAvailable) {
+        if (!isMobileApiAvailable) {
           this.baseUrl = `http://localhost:6927/${this.mw.webUrl.hostname}/v1/page/mobile-sections/`;
         }
-        if (!this.mwCapabilities.veApiAvailable) {
-          this.baseUrlForMainPage = `http://localhost:8000/${this.mw.webUrl.hostname}/v3/page/pagebundle/`;
+
+        if (!isVeApiAvailable) {
+          this.baseUrl = this.mw.mobileRestApiUrl.href;
+          this.baseUrlForMainPage = 'http://localhost:8000/${this.mw.webUrl.hostname}/v3/page/pagebundle/';
         }
       } else {
         logger.log(`Using REST API`);
       }
-    } else {
-      logger.log(`Using remote MCS/Parsoid`);
     }
 
+    logger.log('Base Url: ', this.baseUrl, '\n', 'Base Url for Main Page: ', this.baseUrlForMainPage)
+  }
+
+  public async checkApiAvailabilty(url: string): Promise<boolean>{
+    try {
+      const apiResponse = await this.getJSON<any>(`${url}${encodeURIComponent(this.mw.metaData.mainPage)}`);
+      return !!apiResponse || !!apiResponse.lead || !!apiResponse.visualeditor.content;
```

Comment:
You don't rely on the HTTP status code. Could you please tell me which URL returns a HTTP 200 code even if the API is not available?